keanu-project/src/test/java/io/improbable/keanu/e2e/regression/RadonHierarchicalRegression.java

```diff
@@ -1,181 +1,213 @@
             withPriorOnIntercept(0., 5.).
             build();
-
-        return model;
     }
 
-    private void buildAndRunHeirarchicalNetwork(List<Data> radonData, int numberOfModels) {
-        GaussianVertex muAlpha = new GaussianVertex(new long[]{1, 1}, 0, 100).setLabel("MuIntercept");
-        GaussianVertex muBeta = new GaussianVertex(new long[]{1, 1}, 0, 100).setLabel("MuGradient");
-
-        HalfGaussianVertex sigmaAlpha = new HalfGaussianVertex(new long[]{1, 1}, 10.).setLabel("SigmaIntercept");
-        HalfGaussianVertex sigmaBeta = new HalfGaussianVertex(new long[]{1, 1}, 10.).setLabel("SigmaGradient");
+    private void buildAndRunHierarchicalNetwork(Map<String, List<Data>> radonData, int numberOfModels) {
+        GaussianVertex muAlpha = new GaussianVertex(new long[]{1, 1}, 0, 10).setLabel("MuIntercept");
+        GaussianVertex muBeta = new GaussianVertex(new long[]{1, 1}, 0, 10).setLabel("MuGradient");
 
-        int numPartitions = radonData.size() / numberOfModels;
+        HalfGaussianVertex sigmaAlpha = new HalfGaussianVertex(new long[]{1, 1}, .5).setLabel("SigmaIntercept");
+        HalfGaussianVertex sigmaBeta = new HalfGaussianVertex(new long[]{1, 1}, .5).setLabel("SigmaGradient");
 
-        double[] allFloor = radonData.stream().mapToDouble(k -> k.floor).toArray();
-        double[] allRadon = radonData.stream().mapToDouble(k -> k.log_radon).toArray();
+        List<String> counties = radonData.keySet().stream().sorted().collect(Collectors.toList());
 
-        List<RegressionModel> models = new ArrayList<>();
+        final List<RegressionModel> models = IntStream.range(0, Math.min(numberOfModels, counties.size()))
+            .boxed()
+            .map(i -> radonData.get(counties.get(i)))
+            .map(countyData -> createSubModel(countyData, muBeta, muAlpha, sigmaBeta, sigmaAlpha))
+            .collect(Collectors.toList());
 
-        for (int i = 0; i < numberOfModels; i++) {
-            RegressionModel model = createSubModel(allFloor, allRadon, i, numPartitions, muBeta, muAlpha, sigmaBeta, sigmaAlpha);
-            models.add(model);
-        }
-
-        //Set the starting values of the parent random variables
-        //This is required for the optimiser to find the correct values of the sub models latents
-        muAlpha.setValue(1.);
-        sigmaAlpha.setValue(0.5);
+        // Initialise hierarchical model with linear regression inferred parameters
+        RegressionModel linearRegressionModel = linearRegression(radonData);
+        assertLinearRegressionIsCorrect(linearRegressionModel);
+        muAlpha.setValue(linearRegressionModel.getInterceptVertex().getValue().scalar());
+        muBeta.setValue(linearRegressionModel.getWeightVertex().getValue().scalar());
 
-        muBeta.setValue(-1.);
-        sigmaBeta.setValue(0.5);
+        BayesianNetwork net = new BayesianNetwork(muAlpha.getConnectedGraph());
+        optimise(net); // Fine-tune starting position
```

Comment:
How much does this change the values of the latent variables?

Reply:
A pretty small amount ~0.1 but this does seem to make a difference.

Will take a bit more investigating to understand the root causes behind this stuff.